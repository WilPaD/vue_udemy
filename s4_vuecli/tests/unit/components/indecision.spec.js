import Indecision from '@/components/Indecision'
import { shallowMount, mount } from '@vue/test-utils'

describe('Componente Indecision', () => {

    let wrapper,
        clgSpy;
    beforeEach(() => {
        wrapper = shallowMount( Indecision )

        clgSpy = jest.spyOn( console, 'log' )
    })
    /*
    test('Debe hacer match con snapshot', () => {

        expect( wrapper.html ).toMatchSnapshot()

    })
    */
    test('Escribir en el input no debe disparar nada (console.log)', async() => {

        const getAnswerSpy = jest.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')
        await input.setValue('Hola Mundo')

        //Esperamos que el console log fue llamado 1 vez
        expect( clgSpy ).toHaveBeenCalled(1)
        //Verificamos que la función getAnswer no fué llamada
        expect( getAnswerSpy ).toHaveBeenCalled(0)

    })

    test('Escribir el simbolo "?" debe disparar el fetch', () => {

    })

    test('Pruebas en getAnswer', () => {

    })

    test('pruebas en getAnswer - Fallo en el API', () => {

    })


})