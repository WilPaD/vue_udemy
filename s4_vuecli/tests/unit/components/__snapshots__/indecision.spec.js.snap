// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Indecision Debe hacer match con snapshot 1`] = `[Function]`;

exports[`Componente Indecision Match con snapshot 1`] = `[Function]`;
